src/components/include/utils/threads/thread.h

```diff
@@ -114,6 +114,7 @@ class Thread {
   // Should be locked to protect isThreadRunning_ and thread_created_ values
   sync_primitives::Lock state_lock_;
   volatile bool isThreadRunning_;
+  volatile bool do_iteration_;
```

Comment:
@anosach-luxoft, you are right. It's something that I noticed also, even discussed with @pestOO in some conversation. Frankly, I see many other things to be improved, for example APPLINK-26865, APPLINK-26867.
Should I try to fix (refactor) as many as I can within this ticket or put this ideas in backlog? (I didn't estimated at the beginning further refactoring)